### PR TITLE
[dailymotion] fix extraction

### PR DESCRIPTION
**Pull request by mrfade**
## Please follow the guide below

- You will be asked some questions, please read them **carefully** and answer honestly
- Put an `x` into all the boxes [ ] relevant to your *pull request* (like that [x])
- Use *Preview* tab to see how your *pull request* will actually look like

---

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [adding new extractor tutorial](https://github.com/rg3/youtube-dl#adding-support-for-a-new-site) and [youtube-dl coding conventions](https://github.com/rg3/youtube-dl#youtube-dl-coding-conventions) sections
- [x] [Searched](https://github.com/rg3/youtube-dl/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8)

### In order to be accepted and merged into youtube-dl each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [x] Bug fix
- [ ] Improvement
- [ ] New extractor
- [ ] New feature

---

### Description of your *pull request* and other information

fix issue #17699

### CODE REVIEW

**Comment by randombit** [comment]
This patch works for me, thank you very much!

**Comment by selvodka** [comment]
Works for me too . Thanks :)

**Comment by dhowe** [comment]
How to test this patch?

**Comment by Mehmet4226** [comment]
how to install? i don't install

File: youtube_dl/extractor/dailymotion.py [hunk 1]
```diff
@@ -171,8 +171,7 @@ def _real_extract(self, url):
              r'__PLAYER_CONFIG__\s*=\s*({.+?});'],
             webpage, 'player v5', default=None)
         if player_v5:
-            player = self._parse_json(player_v5, video_id)
-            metadata = player['metadata']
+            metadata = self._download_json('http://www.dailymotion.com/player/metadata/video/%s?integration=inline&GK_PV5_NEON=1' % video_id, video_id)
```
Comment (dstftw) [inline youtube_dl/extractor/dailymotion.py]: 1. This must only take place when it's not available from player JSON.
2. Query must be passed as `query`.

**Comment by Maxou44** [comment]
It works, can you merge it?

File: youtube_dl/extractor/dailymotion.py [hunk 2]
```diff
@@ -172,7 +173,12 @@ def _real_extract(self, url):
             webpage, 'player v5', default=None)
         if player_v5:
             player = self._parse_json(player_v5, video_id)
```
Comment (dstftw) [inline youtube_dl/extractor/dailymotion.py]: Must not be fatal.

File: youtube_dl/extractor/dailymotion.py [hunk 3]
```diff
@@ -24,6 +24,7 @@
     str_to_int,
     unescapeHTML,
     urlencode_postdata,
+    try_get,
```
Comment (dstftw) [inline youtube_dl/extractor/dailymotion.py]: Alphabetic.

File: youtube_dl/extractor/dailymotion.py [hunk 4]
```diff
@@ -172,7 +173,12 @@ def _real_extract(self, url):
             webpage, 'player v5', default=None)
         if player_v5:
             player = self._parse_json(player_v5, video_id)
-            metadata = player['metadata']
+            metadata = try_get(
+                player, lambda x: x['metadata'], dict) or self._download_json(
+                'http://www.dailymotion.com/player/metadata/video/%s' % video_id, video_id, query={
```
Comment (dstftw) [inline youtube_dl/extractor/dailymotion.py]: This URL is available from JSON.

**Pull request closed without merging**